stdlib/json.gr

```diff
@@ -2069,3 +2083,242 @@ provide let parse: (str: String) => Result<Json, JsonParseError> = (str: String)
+ * 
+ * @since v0.6.7
+ */
+provide let isNull = json => {
+  json == JsonNull
+}
+
+/**
+ * Accesses the value of a JSON object property.
+ * 
+ * @param propertyName: The property name to access
+ * @param json: The JSON value
+ * @returns `Ok(value)` containing the property value at the given property name, `Err(Json.PropertyNotFound)` if it is not found, or `Err(Json.NotJsonObject)` if the JSON value is not an object
+ * 
+ * @example
+ * let json = Json.JsonObject([("prop", Json.JsonNumber(123))])
+ * assert Json.get("prop", json) == Ok(Json.JsonNumber(123))
+ * @example
+ * let json = Json.JsonObject([("prop", Json.JsonNumber(123))])
+ * assert Json.get("oops", json) == Err(Json.PropertyNotFound)
+ * @example assert Json.get("prop", Json.JsonNumber(123)) == Err(Json.NotJsonObject)
+ * 
+ * @since v0.6.7
+ */
+provide let get = (propertyName, json) => {
+  match (json) {
+    JsonObject(props) => {
+      match (List.find(((k, _)) => k == propertyName, props)) {
+        Some((_, v)) => Ok(v),
+        None => Err(PropertyNotFound),
+      }
+    },
+    _ => Err(NotJsonObject),
+  }
+}
+
+/**
+ * Accesses the value of a JSON object property, with a default value backup if the property is not found.
+ * 
+ * @param propertyName: The property name to access
+ * @param default: The default value backup to use in the case that the property does not exist on the object
+ * @param json: The JSON value
+ * @returns `Ok(value)` containing the property value at the given property name, `Ok(default)` if the property does not exist on the object, or `Err(Json.IncorrectType)` if the JSON value is not an object
+ * 
+ * @example
+ * let json = Json.JsonObject([("prop", Json.JsonNumber(123))])
+ * assert Json.getOrDefault("prop", json) == Ok(Json.JsonNumber(123))
+ * @example
+ * let json = Json.JsonObject([("prop", Json.JsonNumber(123))])
+ * assert Json.getOrDefault("oops", json) == Ok(Json.JsonNull)
+ * @example
+ * let json = Json.JsonObject([("prop", Json.JsonNumber(123))])
+ * assert Json.getOrDefault("oops", default=Json.JsonNumber(0), json) == Ok(Json.JsonNumber(0))
+ * @example assert Json.getOrDefault("prop", Json.JsonNumber(123)) == Err(Json.IncorrectType)
+ * 
+ * @since v0.6.7
+ */
+provide let getOrDefault = (propertyName, default=JsonNull, json) => {
+  match (get(propertyName, json)) {
+    Ok(x) => Ok(x),
+    Err(PropertyNotFound) => Ok(default),
+    Err(NotJsonObject) => Err(IncorrectType),
+  }
+}
+
+/**
+ * Retrieves the value of a nested JSON object property, recursively accessing
+ * the given property names in order.
+ * 
+ * @param propertyNames: The property names to traverse on the JSON value
+ * @param json: The JSON value
+ * @returns `Ok(value)` containing the property value at the given property access path, `Err({ propertyAccess, error: Json.PropertyNotFound })` if it is not found at some point in the path, or `Err({ propertyAccess, error: Json.NotJsonObject })` if the JSON value is not an object somewhere in the path
```

Comment:
We don't normally document the `Err` contents liket his we normally reference the type on the error.